### PR TITLE
Reduce the public interface and add some integration tests.

### PR DESCRIPTION
**Pull request by ksubrama**

### CODE REVIEW

**Comment by ksubrama** [comment]
/cc @schisamo @tyler-ball 
Add some integration tests first to actually verify that this stuff works correctly for 32-bit.  There were some missing libraries that I had to contend with but it seems to be sorted out now.

I still haven't written chefspec tests yet - that's up next.

File: Berksfile [hunk 1]
```diff
@@ -1,3 +1,11 @@
+def fixture(name)
+  cookbook "mingw_#{name}", path: "test/fixtures/cookbooks/mingw_#{name}"
+end
+
 source 'https://supermarket.chef.io'
 
 metadata
+
+group :integration do
+	fixture 'tdm_install'
```
Comment (schisamo) [inline Berksfile]: Indention looks weird here.

**Comment by schisamo** [comment]
@ksubrama Overall this looks good. I would like @tas50 to add some of us default cookbook engineering things to the repo. Things like a `.travis.yml` which does linting and unit testing.

**Comment by schisamo** [comment]
@ksubrama We should add `resource_name` name entries to your custom resources also.

File: resources/get.rb [hunk 2]
```diff
@@ -26,7 +26,7 @@
 default_action :install
```
Comment (schisamo) [inline resources/get.rb]: We should add `resource_name` name entries to your custom resources also.

File: test/fixtures/cookbooks/mingw_tdm_install/recipes/install64.rb [hunk 3]
```diff
@@ -16,9 +16,9 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-include_recipe 'seven_zip::default'
+include_recipe 'mingw::default'
 
-root_path = node['mingw']['path64']
+root_path = "#{ENV['SYSTEMDRIVE']}\\mingw64"
```
Comment (schisamo) [inline test/fixtures/cookbooks/mingw_tdm_install/recipes/install64.rb]: We could make the `root` attribute of the `mingw_tdm_gcc` resource support a nice default which is based on the  attribute `flavor`. Take a look at what we did in the languages cookbook:
https://github.com/chef-cookbooks/languages/blob/master/libraries/_prefix.rb
https://github.com/chef-cookbooks/languages/blob/master/libraries/language_install.rb#L28
https://github.com/chef-cookbooks/languages/blob/master/libraries/language_execute.rb#L29

File: Berksfile (hunk 1 above)
Comment (ksubrama) [inline Berksfile]: I have got to set :expandtab for these files in vim...

File: test/fixtures/cookbooks/mingw_tdm_install/recipes/install64.rb (hunk 3 above)
Comment (ksubrama) [inline test/fixtures/cookbooks/mingw_tdm_install/recipes/install64.rb]: Agreed - but I still need to explicitly set the root of mingw_get for now - so it doesn't save me that much.  Once we're happy with some layout and I'm sure that we won't need to screw with things, then I'll go add some defaults.

**Comment by ksubrama** [comment]
I forgot to actually enable the travis build.  Enabled now.

**Pull request merged**